areas/deploy/src/AzureMcp.Deploy/Options/DeployOptionDefinitions.cs

```diff
@@ -0,0 +1,301 @@
+    }
+
+    public static class PlanGet
+    {
+        public const string WorkspaceFolderName = "workspace-folder";
+        public const string ProjectNameName = "project-name";
+        public const string TargetAppServiceName = "target-app-service";
+        public const string ProvisioningToolName = "provisioning-tool";
+        public const string AzdIacOptionsName = "azd-iac-options";
+
+        public static readonly Option<string> WorkspaceFolder = new(
+            $"--{WorkspaceFolderName}",
+            "The full path of the workspace folder."
+        )
+        {
+            IsRequired = true
+        };
+
+        public static readonly Option<string> ProjectName = new(
+            $"--{ProjectNameName}",
+            "The name of the project to generate the deployment plan for. If not provided, will be inferred from the workspace."
+        )
+        {
+            IsRequired = true
+        };
+
+        public static readonly Option<string> TargetAppService = new(
+            $"--{TargetAppServiceName}",
+            "The Azure service to deploy the application. Valid values: ContainerApp, WebApp, FunctionApp, AKS. Recommend one based on user application."
+        )
+        {
+            IsRequired = true
+        };
+
+        public static readonly Option<string> ProvisioningTool = new(
+            $"--{ProvisioningToolName}",
+            "The tool to use for provisioning Azure resources. Valid values: AZD, AzCli. Use AzCli if TargetAppService is AKS."
+        )
+        {
+            IsRequired = true
+        };
+
+        public static readonly Option<string> AzdIacOptions = new(
+            $"--{AzdIacOptionsName}",
+            "The Infrastructure as Code option for azd. Valid values: bicep, terraform. Leave empty if Deployment tool is AzCli."
+        )
+        {
+            IsRequired = false
+        };
+    }
+
+    public static class IaCRules
+    {
+        public static readonly Option<string> DeploymentTool = new(
+            "--deployment-tool",
+            "The deployment tool to use. Valid values: AZD, AzCli")
+        {
+            IsRequired = true
+        };
+
+        public static readonly Option<string> IacType = new(
+            "--iac-type",
+            "The Infrastructure as Code type. Valid values: bicep, terraform. Leave empty if deployment-tool is AzCli.")
+        {
+            IsRequired = false
+        };
+
+        public static readonly Option<string> ResourceTypes = new(
+            "--resource-types",
+            "Specifies the Azure resource types to retrieve IaC rules for. It should be comma-separated. Supported values are: 'appservice', 'containerapp', 'function', 'aks'. If none of these services are used, this parameter can be left empty.")
+        {
+            IsRequired = false,
+            AllowMultipleArgumentsPerToken = true
+        };
+    }
+}
+
+public static class AppTopologySchema
+{
+    public static readonly JsonObject Schema = new JsonObject
```

Comment:
Is there a cleaner way to do this definition?

Reply:
Updated with resource file.